go/vm/lfvm/converter.go

```diff
@@ -13,74 +13,56 @@
 package lfvm
 
 import (
-	"bytes"
 	"errors"
 	"fmt"
-	"log"
 	"strings"
-	"sync"
 
 	"github.com/Fantom-foundation/Tosca/go/ct/common"
 	"github.com/Fantom-foundation/Tosca/go/vm"
+	lru "github.com/hashicorp/golang-lru/v2"
 
 	// This is only imported to get the EVM opcode definitions.
 	// TODO: write up our own op-code definition and remove this dependency.
 
 	evm "github.com/ethereum/go-ethereum/core/vm"
 )
 
-type cache_val struct {
-	oldCode []byte
-	code    Code
-}
+const codeCacheCapacity = 50_000 // up to ~ 1 GB for 22 KiB max code size
+
+var cache *lru.Cache[vm.Hash, Code]
 
-var mu = sync.Mutex{}
-var cache = map[vm.Hash]cache_val{}
+func init() {
+	res, err := lru.New[vm.Hash, Code](codeCacheCapacity)
+	if err != nil {
+		panic(fmt.Errorf("failed to create cache: %v", err))
+	}
+	cache = res
+}
 
 func clearConversionCache() {
-	mu.Lock()
-	defer mu.Unlock()
-	cache = map[vm.Hash]cache_val{}
+	cache.Purge()
 }
 
 func Convert(code []byte, with_super_instructions bool, create bool, noCodeCache bool, codeHash vm.Hash) (Code, error) {
```

Comment:
I am a little confused by the name `create`, at first sight I thought that the cache entry is created when it is true. But the opposite is the case.

Reply:
perhaps `isCreateCall` would be a better name  accompanied by the comment below `// Do not cache use-once code in create calls.`

Reply:
Rename to `isInitCode` and fixed the name of the `with_super_instructions` parameter as well.